test/powershell/Language/Parser/Parsing.Tests.ps1

```diff
@@ -423,3 +423,35 @@ Describe "Ternary Operator parsing" -Tags CI {
         $expr.IfFalse | Should -BeOfType 'System.Management.Automation.Language.ConstantExpressionAst'
     }
 }
+
+Describe "Keyword 'default' Token parsing" -Tags CI {
+    BeforeAll {
+        $testCases_basic = @(
+            @{ Script = 'switch (1) {default {0} 1 {1}}'; TokensToCheck = @{ 5 = [System.Management.Automation.Language.TokenKind]::Default; } }
+            @{ Script = 'switch (1) {"default" {0} 1 {1}}'; TokensToCheck = @{ 5 = [System.Management.Automation.Language.TokenKind]::StringExpandable; } }
+            @{ Script = 'switch (1) {adefault {0} 1 {1}}'; TokensToCheck = @{ 5 = [System.Management.Automation.Language.TokenKind]::Identifier; } }
+            @{ Script = 'echo default'; TokensToCheck = @{ 1 = [System.Management.Automation.Language.TokenKind]::Generic; } }
+        )
+    }
+
+    AfterAll {
+    }
+
+    It "'default' correctly tokenized." -TestCases $testCases_basic {
+        param($Script, $TokensToCheck)
+
+        $tks = $null
+        $ers = $null
+        $result = [System.Management.Automation.Language.Parser]::ParseInput($Script, [ref]$tks, [ref]$ers)
+
+        foreach ($token in $TokensToCheck.Keys ) {
+            $tks[$token].Kind | Should -BeExactly $TokensToCheck[$token]
+        }
+    }
+
+    It "'default' can be used as command name" {
+        function default { 'default' }
+
+        default | Should -BeExactly 'default'
```

Comment:
This test seems to be redundant since we already have tests for this case.

Reply:
I cannot find anywhere that this is specifically tested, and I am not really looking in the C# tests, so I may be missing it, either way, I do not know if there is value in this test or not.

Reply:
Perhaps @vexx32 (who pulled #9812) could correct me.

Reply:
I don't think this is quite the same case here. Those tests ensure that we can name a class method after a language keyword, which is a different context than this case — ensuring that a keyword is only valid in a constrained context, and is permitted as a command name outside that context.

I think this test is worthwhile. We don't have many constrained keywords (outside of the aforementioned exception where most keywords are permitted as class method names regardless of their usual restrictions).

Reply:
@vexx32 Many thanks! I was sure that we already have these tests.

@msftrncs Could you please enhance your test with Loel's suggesion? (TestCases helps.)

Reply:
I will reorganize the tests that I was adding here to be a more generic set of tests for 'limited scope' keywords.  I will include `in` as well, as its also a limited scope keyword, if I do not find any tests for it.